src/masternode-budget.cpp

```diff
@@ -110,24 +110,24 @@ bool IsBudgetCollateralValid(const uint256& nTxCollateralHash, const uint256& nE
 
 void CBudgetManager::CheckOrphanVotes()
 {
-    LOCK(cs);
-
     std::string strError = "";
-    auto it1 = mapOrphanMasternodeBudgetVotes.begin();
-    while (it1 != mapOrphanMasternodeBudgetVotes.end()) {
-        if (budget.UpdateProposal(((*it1).second), NULL, strError)) {
-            LogPrint(BCLog::MNBUDGET,"%s: Proposal/Budget is known, activating and removing orphan vote\n", __func__);
-            mapOrphanMasternodeBudgetVotes.erase(it1);
+    {
+        LOCK(cs_votes);
+        for (auto it = mapOrphanMasternodeBudgetVotes.begin(); it != mapOrphanMasternodeBudgetVotes.end();) {
+            if (UpdateProposal(it->second, nullptr, strError))
+                it = mapOrphanMasternodeBudgetVotes.erase(it);
```

Comment:
In a future work, would be nice to re organize / re think the proposal update, request and removal on this method.
As is now, `CheckOrphanVotes` locks `cs_votes`, calls `UpdateProposal` which locks `cs_proposals` and locks one more time `cs_votes`.

Reply:
Yup.
`Update` flow fully restructured in round 3  (#1845).